### PR TITLE
Todo Lists In JavaScript

### PR DESCRIPTION
**Pull request by karthykarthick**
In this user can
- create, 
- read, 
- update,
-  delete lists it will save local storage.

### CODE REVIEW

File: src/components/Todos.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        this.changeValues(todoIndex);
+      });
+    });
+  }
+
+  changeValues(idx) {
+    const editBtn = document.querySelector('.edit-form-btn');
+    editBtn.addEventListener('click', () => {
+      // e.preventDefault();
```
Comment (Torres-ssf) [inline src/components/Todos.js]: - [x] We need to remove all commented code.

File: src/components/Ui.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            <li class="list-group-item">priority: ${todo.priority}</li>
+          </ul>
+          <button class="btn btn-outline-danger del-data" data-name="${todo.title}">Delete</button>
+          <button class="btn btn-outline-success complete-btn" data-sucess="${todo.title}">Completed task</button>
+          <button class="btn btn-outline-warning edit-btn" data-edit="${sanitizeName(todo.title)}">Edit</button>
+        </div>
+      </div>
+        `;
+      this.todosConent.insertAdjacentHTML('afterbegin', todoCard);
+      // todos.completedTask()
```
Comment (Torres-ssf) [inline src/components/Ui.js]: - [x] We need to remove all commented code

File: package.json [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
   },
   "keywords": [],
   "author": "",
   "license": "ISC",
   "bugs": {
-    "url": "https://github.com/kcotrinam/webpackTemplate/issues"
+    "url": "https://github.com/karthykarthick/To-Do-List-Js/issues"
   },
-  "homepage": "https://github.com/kcotrinam/webpackTemplate#readme",
+  "homepage": "https://github.com/karthykarthick/To-Do-List-Js#readme",
```
Comment (Torres-ssf) [inline package.json:23]: - [x] The homepage property is duplicated at the package.json file.

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi team :hand: 
Awesome job on this project so far :clap: 
But we need to make some changes before moving on.

- [x] You did a nice job setting up the linter on the PR using Github actions. However, it was not properly set up in your repository. We need to follow the instructions [here](https://github.com/microverseinc/linters-config/tree/master/javascript) and add `eslint` and it's needed dependencies as devDependencies. The linter should be used in the development stages, It helps us developers to follow good practice and to maintain clean code.

- [x] We need to remove all `eslint-disable` commented lines from your code to make it accordingly with the linter rules. Instead of disabling it, we need to use the recommended approach. You can google it to learn more about the rule and how you can solve the error. [Example](https://eslint.org/docs/rules/linebreak-style). Setting up `eslint` on your repo it will help you to avoid these errors.

- [ ] As the requirements said, the application should have a default Project already created.

Happy coding :computer: 
**Sergio Torres**

**Comment by bellom** [comment]
> ### Changes required 💪
> Hi team ✋
> Awesome job on this project so far 👏
> But we need to make some changes before moving on.
> 
> * [x]  You did a nice job setting up the linter on the PR using Github actions. However, it was not properly set up in your repository. We need to follow the instructions [here](https://github.com/microverseinc/linters-config/tree/master/javascript) and add `eslint` and it's needed dependencies as devDependencies. The linter should be used in the development stages, It helps us developers to follow good practice and to maintain clean code.
> * [x]  We need to remove all `eslint-disable` commented lines from your code to make it accordingly with the linter rules. Instead of disabling it, we need to use the recommended approach. You can google it to learn more about the rule and how you can solve the error. [Example](https://eslint.org/docs/rules/linebreak-style). Setting up `eslint` on your repo it will help you to avoid these errors.
> * [ ]  As the requirements said, the application should have a default Project already created.
> 
> Happy coding 💻
> **Sergio Torres**

@karthykarthick @kcotrinam, I think you guys miss the second comment from @Torres-ssf.
Kindly work on it.

File: src/components/Project.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-unused-vars */
+/* eslint-disable linebreak-style */
+// eslint-disable-next-line linebreak-style
+/* eslint-disable import/no-cycle */
+/* eslint-disable linebreak-style */
```
Comment (bellom) [inline src/components/Project.js]: Please remove ESLint disabling rules comments

File: src/components/Todos.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable max-len */
+/* eslint-disable linebreak-style */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable linebreak-style */
+/* eslint-disable import/no-cycle */
+/* eslint-disable linebreak-style */
```
Comment (bellom) [inline src/components/Todos.js]: Please remove ESLint disabling rules comments

File: src/components/Ui.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+/* eslint-disable no-use-before-define */
+/* eslint-disable array-callback-return */
+/* eslint-disable linebreak-style */
+/* eslint-disable eqeqeq */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable linebreak-style */
+/* eslint-disable import/no-mutable-exports */
+/* eslint-disable linebreak-style */
+/* eslint-disable import/no-cycle */
+/* eslint-disable linebreak-style */
```
Comment (bellom) [inline src/components/Ui.js]: Please remove ESLint disabling rules comments

File: src/components/storage.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable import/no-cycle */
+/* eslint-disable linebreak-style */
```
Comment (bellom) [inline src/components/storage.js]: Please remove ESLint disabling rules comments

File: src/index.js [hunk 8]
```diff
@@ -1 +1,44 @@
-import './scss/styles.scss';
\ No newline at end of file
+/* eslint-disable linebreak-style */
+/* eslint-disable no-unused-vars */
+/* eslint-disable linebreak-style */
+/* eslint-disable import/no-cycle */
+/* eslint-disable linebreak-style */
```
Comment (bellom) [inline src/index.js]: Please remove ESLint disabling rules comments

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

**According to @Torres-ssf, you have told to remove Eslint inline comment, kindly attend to it.**

**_In other words;_**
You are not modifying linter config files directly, but you are disabling many ESLint rules with the inline Eslint comments in your codebase, and that's kind of bypassing ESLint configuration which represent style guidelines that you share with your team - which is a group of all Microverse students. (screenshot from Pathwright).

 Please remove ESLint disabling rules comments and fix your code accordingly.
I'm sure this is not going to be an issue for you, you got this 💪.

Remember that you can run ESLint locally to debug the issues and fix them before pushing, and you probably have already installed the VSCode ESLint extension if you're using VSCode, which can be also very helpful.

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello Team 👋

**You've done pretty well here fixing a few issues raised in the previous review. 👌**

However, let's make a couple of changes.

- [ ] Kindly link the `index.js` in the `index.html` this provides a way to run the project after bundling the modules with webpack. 

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/components/Todos.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+//   const btns = document.querySelectorAll('.complete-btn');
+
+//   btns.forEach(btn => {
+//     btn.addEventListener('click', e => {
+//       const card = document.querySelector(`#${e.target.dataset.sucess}`);
+//       card.classList.toggle('border');
+//       card.classList.toggle('border-success');
+//     });
+//   });
+// };
```
Comment (mikenath223) [inline src/components/Todos.js]: - [x] Please remove this commented code block. Kindly use git history if you want to refer to previous code versions.

File: src/components/Ui.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+          <button class="btn btn-outline-success complete-btn" data-sucess="${todo.title}">Completed task</button>
+          <button class="btn btn-outline-warning edit-btn" data-edit="${sanitizeName(todo.title)}">Edit</button>
+        </div>
+      </div>
+        `;
+      this.todosConent.insertAdjacentHTML('afterbegin', todoCard);
+    });
+    const todos = new Todo();
+    todos.deleteButton();
+    // todos.completedTask();
```
Comment (mikenath223) [inline src/components/Ui.js]: - [x] Also kindly remove this commented code.

**Review by Torres-ssf — APPROVED** [review]
### PR Approved :+1: 
Hello again team :hand: 
You may merge the PR
Good luck with the next project

Happy coding :computer: 
**Sérgio Torres**

**Pull request merged**